crates/db_views/src/post_view.rs

```diff
@@ -337,17 +334,15 @@ impl<'a> PostQuery<'a> {
     }
     // Only hide the read posts, if the saved_only is false. Otherwise ppl with the hide_read
     // setting wont be able to see saved posts.
-    else if !self.show_read_posts.unwrap_or(true) {
+    else if !self.local_user.map(|l| l.show_read_posts).unwrap_or(true) {
       query = query.filter(post_read::id.is_null());
     }
 
-    // Filter out the rows with missing languages
-    if self.my_local_user_id.is_some() {
+    if self.local_user.is_some() {
+      // Filter out the rows with missing languages
       query = query.filter(local_user_language::id.is_not_null());
-    }
 
-    // Don't show blocked communities or persons
-    if self.my_person_id.is_some() {
+      // Don't show blocked communities or persons
       query = query.filter(community_block::person_id.is_null());
       query = query.filter(person_block::person_id.is_null());
     }
```

Comment:
Please check that this is correct, not entirely sure how it works.

Reply:
Yep its good, the tests would be failing otherwise.